### ISSUE TITLE
Reliable crash in 43.1 from CLAP which generates Note Expressions when piped into non-CLAP

### ISSUE THREAD

**Issue by baconpaul** [free-audio/interop-tracker#37]
Hi!

I have a set of claps which generate note expressions as a note effect and they reliably crash the engine if piped into a device which is not a CLAP which supports note expressions.

Here's how to reproduce

1: grab https://github.com/surge-synthesizer/tuning-note-claps and build the .clap
2: Set up a single track which is the EDMN Clap into the Poly Grid. Use default settings
3: With a midi keyboard, play fast arpeggios. It will crash

If you comment out line 197 of helpers.h (there's a comment right above it) it won't crash. But it also won't work because that suppresses the note expression sends.

If I run the engine in the debugger, the crash isn't in my plug. And with the line commented out my plug doesn't crash. So it really is the act of adding the note expressions to the outbound stream that crashes bitwig clap.

I *think* this is a problem in the out-of-clap Note Expression translation layer. If I do the same configuration (and EDMN Clap into Surge XT Clap) it both tunes correctly and doesn't crash. So something about outbound clap note expressions into bitwig devices blows up stuff.

I don't have an VST3 which support note expression on my box so couldn't test that configuration.

**Comment by baconpaul** [comment]
Yeah if I put the clap saw demo behind the EDMN Note Expression I can also play it indefinitely. This definitely looks like an engine crash when converting CLAP note expressions into (whatever is downstream for poly grid etc...)

**Comment by robbert-vdh** [comment]
If it's useful, the segfault seems to happen in a function called `__NCMonoLogic_process`. If you attach gdb you get: 

```
malloc(): unsorted double linked list corrupted
Exception in thread   bitwig-remote-pluginhost-91157-10
```

**Comment by baconpaul** [comment]
yeah i couldn't get a symbol on macOS. Thanks robbert!

**Comment by robbert-vdh** [comment]
I only quickly tested this with 4.3.0 (since there are some windowing issues with 4.3.1) and both the plugin from the OP as well as [this one](https://github.com/robbert-vdh/nih-plug/tree/master/plugins/examples/midi_inverter), but I'll assume this behavior hasn't changed in 4.3.1 and it's the same crash.

**Comment by abique** [comment]
I looked into it, and it did not crash on me.

**Comment by baconpaul** [comment]
with which version of bitwig on which platform?
every user who tried this other than you has a crash!

**Comment by baconpaul** [comment]
and what configuration of plugin and synth did you use?

clap -> clap works
clap -> polygrid crashes

**Comment by robbert-vdh** [comment]
Also doesn't seem to crash anymore for me on 4.3.2 (tested with the same midi inverter plugin I posted above).

**Comment by baconpaul** [comment]
oooh lemme grab 4.3.2 and test.

**Comment by baconpaul** [comment]
nope can still crash the engine in 10 seconds in 4.3.2

**Comment by baconpaul** [comment]
[NoteExpCrash.zip](https://github.com/free-audio/interop-tracker/files/9161282/NoteExpCrash.zip)

Load this project.
Play 3 notes in a row quickly over and over
i get a crash within 5 seconds in the engine not the plugin

**Comment by robbert-vdh** [comment]
~~I think it only happens with live MIDI. There are no crashes when playing back something like this (with all of the expressions randomized for each note):~~
~~![afbeelding](https://user-images.githubusercontent.com/748520/180274666-7f0f51f2-d2ec-409b-a643-7e8e6d6dd756.png)~~

Never mind, I still had the plugin bypassed.

**Comment by baconpaul** [comment]
right so you get the crash still? with which version and os?

**Comment by baconpaul** [comment]
can you share your crashing project also perhaps?

**Comment by robbert-vdh** [comment]
I used your project as is and with the midi_inverter I linked above, Bitwig 4.3.2 on Linux.

**Comment by baconpaul** [comment]
gotcha. so i wonder if it is somehow fixed in the 4.3.3+ alex is running or if theres a setup issue in his env to repro it?

**Comment by abique** [comment]
Fixed for 4.3.4.

**Comment by baconpaul** [comment]
Awesome!!
